internal/controller/freight/finder.go

```diff
@@ -19,6 +19,7 @@ func FindCommit(
 	desiredOrigin *kargoapi.FreightOrigin,
 	freight []kargoapi.FreightReference,
 	repoURL string,
+	mustFind bool,
```

Comment:
Most of the time, it's a problem if we fail to find the artifact we're looking for. Most callers will set this to true. There are some instances where the caller is more tolerant of something not being found, in which case they can set this to false and deal with the nil artifact themselves.

Reply:
Fixed in c2b1b014